doc/_templates/search.html

```diff
@@ -0,0 +1,28 @@
+{%- extends "!search.html" %}
+{%- block scripts %}
+  {% if google_searchengine_id %}
+  {{ super.super() }}
+    <link rel="stylesheet" href="{{ pathto('_static/css/gcs.css', 1) }}" type="text/css" />
+  {% else %}
+    {{ super() }}
+  {% endif %}
+{%- endblock %}
+
+{% block footer %}
+  {% if google_searchengine_id %}
+    {{ super.super() }}
+  {% else %}
+    {{ super() }}
+  {% endif %}
+{% endblock %}
+
+{% block body %}
+  {% if google_searchengine_id %}
+    <h2>{{ _('Search Results') }}</h2>
+    <script async src="https://cse.google.com/cse.js?cx={{ google_searchengine_id }}">
```

Comment:
Depending on an external partner (one that may indeed be unavailable, have limited access, be biased, not allowed by some companies' infrastructure) is a no-go for our project documentation IMO. Same goes for locally built documentation.

Reply:
Point taken, and also one of the reasons why I'd like to keep it super straightforward to switch back to "self-contained" search. 

Regarding relying on 3rd party services, our code is hosted on GitHub, our CI runs on GitHub and AWS, our docs are hosted on AWS, and a lot of the community interactions happen on Discord, all of which can be unavailable at times, or not be accessible / blocked for some folks :/